sessionctx/sessionstates/session_states_test.go

```diff
@@ -102,11 +102,6 @@ func TestSystemVars(t *testing.T) {
 			varName:         variable.TiDBMaxTiFlashThreads,
 			expectedValue:   strconv.Itoa(variable.DefTiFlashMaxThreads),
 		},
-		{
-			// hidden variable
-			inSessionStates: false,
-			varName:         variable.TiDBTxnReadTS,
-		},
```

Comment:
`TiDBTxnReadTS`  should not be encoded into `sessionStates` for session manager. It is very special that it can be set only once and be consumed. I think we need to consider twice for every hidden variable.

@morgo FYI

Reply:
OK. We can keep them out of session manager, but visible in SHOW VARIABLES. That still works for my purposes.

Reply:
@xhebox PTAL again